lib/matplotlib/tight_bbox.py

```diff
@@ -45,6 +50,9 @@ def restore_bbox():
         fig.set_tight_layout(orig_tight_layout)
         fig.transFigure._boxout = _boxout
         fig.transFigure.invalidate()
+        for l in fig.legends:  # re-set the anchor for legends...
+            anch = legend_anchors[l].transformed(fig.transFigure.inverted())
+            l.set_bbox_to_anchor(anch)
```

Comment:
This kind of assumes what the original `transform` argument was, doesn't it?

Also, the setter will set the artist `stale`. Maybe not using the setter is better?

Reply:
Yep, it assumes no transform.  Does the transform argument work at all?

Reply:
There's one example in the constrained layout guide:
https://github.com/matplotlib/matplotlib/blob/9ccdb1ddcd6250fac950ddcab97c41f59f68dd72/tutorials/intermediate/constrainedlayout_guide.py#L246-L247

Reply:
Ohhhh, I see.   OK this is a much easier and more robust fix.